bcda/auth/ssas.go

```diff
@@ -14,9 +16,31 @@ type SSASPlugin struct {
 }
 
 // RegisterSystem adds a software client for the ACO identified by localID.
-func (s SSASPlugin) RegisterSystem(localID string) (Credentials, error) {
-	// s.client.CreateSystem()
-	return Credentials{}, errors.New("not yet implemented")
+func (s SSASPlugin) RegisterSystem(localID, publicKey string) (Credentials, error) {
+	creds := Credentials{}
+	aco, err := GetACOByClientID(localID)
+	if err != nil {
+		return creds, errors.Wrap(err, "failed to create system")
+	}
+	trackingID := uuid.NewRandom().String()
+
+	cb, err := s.client.CreateSystem(
+		aco.Name,
+		*aco.CMSID,
```

Comment:
This assignment assumes that ACO-MS will make the group ids CMS IDs. That is an assumption we cannot make. I think the best choice available to us is to add groupID as a method argument.

Reply:
then would bcda have to store the groupID as well?

Reply:
BCDA probably doesn't need to store it. The current plan is to create a group from the CLI and use the returned group ID when creating the ACO and system.

Reply:
okay. made a change so that BCDA can provide that groupID (instead of using the ACO's CMS ID)